### ISSUE TITLE
README suggestion

### ISSUE THREAD

**Issue by aspiers** [roman-neuhauser/git-pimp#1]
Looks nice!  Especially the comprehensive documentation and test suite.  It's not immediately clear to me from the README exactly what advantages this offers over the vanilla `git format-patch && git send-email` approach, so it would be helpful if the README mentioned that explicitly.

**Comment by roman-neuhauser** [comment]
this is the same situation as in roman-neuhauser/git-mailz/issues/1: constructing the manpage as a list of differences from other software rubs me the wrong way (and i don't want to decouple the readme from the manpage yet).

i guess i need a git-pimp-tutorial(1) manpage, that might even be a better readme.

what do you think?

**Comment by aspiers** [comment]
I agree that the man page is not the right location for a "competitive" analysis, but I also believe that the README is definitely not the right place for the man page.  Why do you want to avoid decoupling them?  That necessarily creates problems, of which this issue is one example.

In general I think a good README should give a quick overview of all the important points regarding the software, and therefore contain the following:
- what it does
- why it exists (including a brief comparison to close alternatives, and ideally a link to another page which gives more detail, especially if the project could be miscontrued as reinventing wheels)
- how to install it (or if that's not simple, a link to the installation guide)
- briefly how to use it
- a link to the full documentation, which in this case is the man page, but could also include tutorials etc.
- where/how to submit bug reports / enhancement requests
- how to contribute to development
- the license (or a link to it)

**Comment by roman-neuhauser** [comment]
> [I believe that] the README is definitely not the right place
> for the man page. Why do you want to avoid decoupling them?

for a tiny program like git-pimp, there's very little to be said
that shouldn't be in the manpage anyway.  i used to do READMEs
in the "traditional" format you're arguing for, and i found that
when they had different but overlapping content i frequently
forgot to update one of them when the change should've affected
both.

the manpage-in-readme approach already satisfied five of the eight points
you've listed, namely
- what it does
- briefly how to use it
- a link to the full documentation, which in this case is the man page,
  but could also include tutorials etc.
- where/how to submit bug reports / enhancement requests
- how to contribute to development

of the remaining points:
- the license can be trivially mentioned in the manpage
- how to install it could be given in a separate INSTALL file

i've done just that

leaving us with
- why it exists (including a brief comparison to close alternatives, and
  ideally a link to another page which gives more detail, especially if the
  project could be miscontrued as reinventing wheels)

which points to the real problem, which is that the manpage is very light
on behavioral details: comparison is unnecessary to people who already know
the alternatives and meaningless to people who don't.

stay tuned for improvements to the manpage.
